src/hooks/useUserGauge.ts

```diff
@@ -95,9 +110,32 @@ export default function useUserGauge(gaugeAddress?: string): UserGauge | null {
     stake: gaugeContract["deposit(uint256)"],
     unstake: gaugeContract["withdraw(uint256)"],
     claim: () => {
-      const promises = [gaugeMinterContract.mint(gaugeAddress)]
-      if (hasExternalRewards) {
-        promises.push(gaugeContract["claim_rewards(address)"](account))
+      const promises = []
+      try {
+        if (hasExternalRewards) {
+          promises.push(gaugeContract["claim_rewards(address)"](account))
+        }
+
+        if (isMainnet(chainId)) {
+          const gaugeMinterContract = getGaugeMinterContract(
+            library,
+            chainId,
+            account,
+          )
+
+          promises.push(gaugeMinterContract.mint(gaugeAddress))
+        } else {
+          const childGaugeFactory = getChildGaugeFactory(
+            library,
+            chainId,
+            registryAddresses["ChildGaugeFactory"],
+            account,
+          )
+          promises.push(childGaugeFactory.mint(gaugeAddress))
+        }
+      } catch (e) {
+        console.error(e)
+        enqueueToast("error", "Unable to claim reward")
       }
 
       return Promise.all(promises)
```

Comment:
@penandlim, can we eliminate the stale veSDL problem by running some kind of checkpoint function in serial before returning this `Promise.all()`?  Potentially `gaugeContract.user_checkpoint(userAddress)`?